app/controllers/teachers_controller.rb

```diff
@@ -42,22 +42,25 @@ def new
     @school = @teacher.school # maybe delegate this
     @readonly = false
     if omniauth_data.present?
+      email_data = { email: omniauth_data.delete("email"), primary: true }
       @teacher.assign_attributes(omniauth_data)
+      @teacher.email_addresses.build(email_data)
     end
   end
 
   # TODO: This needs to be re-written.
   # If you are logged in and not an admin, this should fail.
   def create
     # Find by email, but allow updating other info.
-    @teacher = Teacher.find_by(email: teacher_params[:email])
+    teacher_id = EmailAddress.find_by(email: teacher_params[:primary_email])&.teacher_id
+    @teacher = Teacher.find_by(id: teacher_id)
```

Comment:
If you don't use `_id` then you can get the association record directly:

```suggestion
    @teacher = EmailAddress.find_by(email: teacher_params[:primary_email])&.teacher
```